### PR TITLE
qt: Drop PlatformStyle

### PR DESCRIPTION
**Pull request by xdustinface**
This PR ist part of a series of +-25 PRs related to UI redesigns. Its ancestor is #3572, its successor is  #3574. I did not screenshot every single PR and its changes, instead i made "walk through all screen" videos with the result of this PR series and also with the 0.15 UI. If there are any concrete screenshots wanted, just let me know. To build with the full set of changes you can build from the branch [xdustinface:pr-ui-redesign](https://github.com/xdustinface/dash/tree/pr-ui-redesign) which always contains all changes.

[ -> Walk through 0.15](https://youtu.be/cWQeHj5tWR0)
[ -> Walk through Redesign](https://youtu.be/0QeSyXo1aao)

I tried to give the commits enough text to make things obvious without a lot description for each PR. Also here, if you want more description for this specific PR, let me know.
### About this PR

This PR adds main menu icons to the list of icons affected by "PlatformStyle -> imagesOnButtons". It also disables any of the PlatformStyle related changes. I would propose to remove PlatformStyle code from the codebase and replace it with some new "ThemeManager" implementation. I would take this in the future if we agree on that. Reason why i disabled them is mostly personal pferecence.. for me having those icons in the menu and on buttons looks a bit outdated and on top i would say our icon set has too wildly mixed colors and icon styles which give a weird look. I feel/hope i will find supporter for this? :D

### CODE REVIEW

**Comment by xdustinface** [comment]
Ready for review! 

This are some of the icons affected by this change.

icons2
icons

**Comment by xdustinface** [comment]
@PastaPastaPasta I will ask suba what he thinks about removing them + im anyway currently waiting on a new icon set by him which will be added by #3574 then and we could potentially enable those again as menu icons if it looks good but i still think we should evaluate if it really makes sense to have icons in there or if this is just some "we keep it from the old days" thing? I have really not a lot applications in my windows/linux VMs but the ones i looked at did not have any icons in there. Do you have any?

**Comment by UdjinM6** [comment]
These changes contradict one each other imo. There is no reason to use `platformStyle->Icon()` if you disable platform styles completely and it always translates to `QIcon()`.

**Comment by xdustinface** [comment]
Yeah thats true, the reason i didn't just put an `QIcon()` there in just to have the option to turn them on again in case with the platform styles without a lot changes required. But if we agree on just removing them im totally open to just put an `QIcon()` instead.

**Comment by xdustinface** [comment]
Ok so suba confirmed its ok to remove the icons there. @PastaPastaPasta @UdjinM6 what are your thoughts? Also let me know what you think (in case we decide to remove them) regarding the `platformStyle->Icon()` removal and with it the cleaning up of the rest. I would just remove all calls of `setIcon` from the diff and just create the `QActions` without icons.

**Comment by UdjinM6** [comment]
I'm fine with dropping these icons and `setIcon` calls. But then we can just drop the whole `PlatformStyle` thing completely (instead of disabling it) because it's not going to be used for anything after that.

**Comment by xdustinface** [comment]
@UdjinM6 Im fine with dropping it, done in ade639c! I guess that also means doors are opened to later add the "theme/appearance manager" thing i was talking about to clean all that new UI stuff up a bit and move it out of `GUIUtil`?

File: src/qt/dash.cpp [hunk 1]
```diff
@@ -483,7 +469,7 @@ void BitcoinApplication::initializeResult(bool success)
     if(success)
     {
         // Log this only after AppInitMain finishes, as then logging setup is guaranteed complete
-        qWarning() << "Platform customization:" << platformStyle->getName();
+        qWarning() << "Platform customization:" << gArgs.GetArg("-uiplatform", BitcoinGUI::DEFAULT_UIPLATFORM).c_str();
```
Comment (UdjinM6) [inline src/qt/dash.cpp:472]: This line, the `-uiplatform` option and `BitcoinGUI::DEFAULT_UIPLATFORM` const should be dropped.

Comment (xdustinface) [reply src/qt/dash.cpp:472]: Why? Its still used for the platform specific css style thing we just introduced.

Comment (UdjinM6) [reply src/qt/dash.cpp:472]: Ahhh, right! nvm then :)

**Comment by xdustinface** [comment]
Are we good with dropping the unused icons in #3574 which will anyway contain a update to all used icons?

**Comment by UdjinM6** [comment]
Yes, I guess it makes sense to update/drop them all in one place, good idea.

**Pull request merged**